src/gettod.c

```diff
@@ -463,17 +471,17 @@ int W32_CALL gettimeofday2 (struct timeval *tv, struct timezone *tz)
   {
     static uint64 last = 0;
     static time_t secs = 0;           /* UTC time() at midnight, local time */
-    uint64 usecs = microsec_clock();  /* usec day-clock */
+    uint64 now = microsec_clock();    /* day-clock (fixed-point NTP format) */
 
-    if (secs == 0 || usecs < last)    /* not init or wrapped */
+    if (secs == 0 || now < last)      /* not init or wrapped */
     {
       secs = time (NULL);
-      secs -= usecs / 1000000UL;
+      secs -= now >> 32;
     }
-    last = usecs;
+    last = now;
 
-    tv->tv_usec = usecs % 1000000UL;
-    tv->tv_sec = (usecs - tv->tv_usec) / 1000000UL + secs;
+    tv->tv_usec = (now & 0xffffffffUL) * 1000000UL >> 32;
```

Comment:
When reading this, I wonder what the operator precedence is. Should it be:
 * `tv->tv_usec = (now & 0xffffffffUL) * (1000000UL >> 32);` or:
  * `tv->tv_usec = ((now & 0xffffffffUL) * 1000000UL) >> 32;`

Please make it clear. Ref: https://en.cppreference.com/w/c/language/operator_precedence

Reply:
Good point, will do.